frontstage/views/account/account_survey_share.py

```diff
@@ -0,0 +1,131 @@
+import logging
+
+from flask import render_template, request, flash, url_for
+from flask import session as flask_session
+from structlog import wrap_logger
+from werkzeug.utils import redirect
+
+from frontstage import app
+from frontstage.common.authorisation import jwt_authorization
+from frontstage.controllers import survey_controller
+from frontstage.controllers.party_controller import get_list_of_business_for_party, \
+    get_surveys_listed_against_party_and_business_id, get_business_by_id
+from frontstage.exceptions.exceptions import ShareSurveyProcessError
+from frontstage.models import AccountSurveyShareBusinessSelectForm, AccountSurveyShareRecipientEmailForm, \
+    ConfirmEmailChangeForm
+from frontstage.views.account import account_bp
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+@account_bp.route('/share-surveys', methods=['GET'])
+@jwt_authorization(request)
+def share_survey_overview(session):
+    flask_session['business_selected'] = None
```

Comment:
It seems like these are being set to none too often.  Is it being done to make sure you start the process with a clean slate?  If so, it'd be better to delete the key to keep the session clean instead of setting it to none (and leaving an empty key there).

Also the naming is fine but I think we can do better to make it easier for future devs.  In the session how you have it,  it's basically a dict that looks like:
```python
{
    'business_selected': 'some data..',
    'surveys_selected': ['data'],
    'receipient_email_address': 'an email address',
    ....other session data
}```

it might be better to organise it like:
```python
{
    'share_survey': {
        'business_selected': 'some data..',
        'surveys_selected': ['data'],
        'receipient_email_address': 'an email address',
    }
    ....other session data
}```

this way it's crystal clear what the data is for, and the possibility of name clashes in the future get reduced to nearly zero.

Reply:
having share_survey key as a consolidated group is a good idea. Can do that. Deleting a session key will need to make sure that the key exists in the session. Which will have additional logic everywhere, to avoid that its better to make a value of None rather deleting it.

Reply:
said that, it seems that having a dict as the flask session is not as efficient as it looks and requires more for maintaining it. A key value pair for the session key seems like the right approach. I will provide a prefix as discussed to make sure its not too generic.